### PR TITLE
Fix boolean pattern parse

### PR DESCRIPTION
**Pull request by cmrd-senya**
Fixes the regular expression for booleans

### CODE REVIEW

File: spec/lib/diaspora_federation/entity_spec.rb [hunk 1]
```diff
@@ -195,16 +195,28 @@ module DiasporaFederation
...
+          expect(entity.test).to eq(false)
+        end
+
+        it "parses boolean fields with a randomly matching pattern as erroneous" do
+          %w(ttFFFtt yesFFDSFSDy noDFDSFFDFn fXf LLyes).each do |weird_value|
+            xml = <<-XML.strip
+<test_entity_with_boolean>
+  <test>#{weird_value}</test>
+</test_entity_with_boolean>
+            XML
```
Comment (SuperTux88) [inline spec/lib/diaspora_federation/entity_spec.rb]: Oh, one little thing, can you remove the spaces in this line?

Comment (cmrd-senya) [inline spec/lib/diaspora_federation/entity_spec.rb]: Done.
It must have been my IDE.

Comment (SuperTux88) [inline spec/lib/diaspora_federation/entity_spec.rb]: Yes, it was the IDE, I know that ;) Thanks :)

**Comment by SuperTux88** [comment]
Thanks ❤️

File: lib/diaspora_federation/entity.rb [hunk 2]
```diff
@@ -319,8 +319,8 @@ def simple_node(doc, name, value)
       when :integer
         text.to_i if text =~ /^\d+$/
       when :boolean
-        return true if text =~ /(true|t|yes|y|1)$/i
-        false if text =~ /(false|f|no|n|0)$/i
+        return true if text =~ /^(true|t|yes|y|1)$/i
+        false if text =~ /^(false|f|no|n|0)$/i
```
Comment (jhass) [inline lib/diaspora_federation/entity.rb]: Careful here, in Ruby `^`and `$`are beginning/end of **line** respectively, regardless of `/m`. If Beginning/end of **string** is intended here, use `\A`and `\z`respectively.

Comment (cmrd-senya) [inline lib/diaspora_federation/entity.rb]: Is it a real issue here? We can declare multiline value for booleans, integers and timestamps as undefined behavior. I'm not sure one would send multiline strings here and expect some determined results. Perhaps, we could add a validation for a single-line value instead for these types.

Comment (jhass) [inline lib/diaspora_federation/entity.rb]: I wouldn't even discuss as much what makes sense or not on the protocol level, it should be a coding style rule to use `\A`/`\z`unless one explicitly wants the line-wise behavior.

Comment (cmrd-senya) [inline lib/diaspora_federation/entity.rb]: opened #47

Comment (SuperTux88) [inline lib/diaspora_federation/entity.rb]: I agree with @jhass here, it's better codestyle and I also used `\A`/`\z` at other places, but not everywhere (so it's my fault), so we should do it everywhere unless we need the line behavior. :) Thanks for your PR.

**Pull request merged**